wgpu-native/src/binding_model.rs

```diff
@@ -36,6 +37,7 @@ pub struct BindGroupLayoutDescriptor {
 
 pub struct BindGroupLayout<B: hal::Backend> {
     pub(crate) raw: B::DescriptorSetLayout,
+    pub(crate) bindings: Vec<BindGroupLayoutBinding>,
```

Comment:
It is possible to have a slice here instead?

Reply:
Do you mean `Box<[BindGroupLayoutBinding]>`? or an actual slice? The slice would require borrowing, while this structure is owned. I'll use `ArrayVec` on the `PipelineLayout` as an optimization, but not sure how to improve this one.